### PR TITLE
(PUP-10610) Change disable_i18n setting to true by default

### PR DESCRIPTION
**Pull request by luchihoratiu**
Starting with Puppet 7, the `disable_i18n` setting changes its default value from `false` to `true`. Locales are also not pluginsynced anymore when i18n is disabled.

### CODE REVIEW

**Comment by joshcooper** [comment]
We should also skip the `locales` pluginsync if `Puppet[:disable_i18n]` is true. https://github.com/puppetlabs/puppet/blob/main/lib/puppet/configurer/plugin_handler.rb#L32-L51

File: spec/unit/node/environment_spec.rb [hunk 1]
```diff
@@ -526,34 +526,55 @@ def assert_manifest_conflict(expectation, envconf_manifest_value)
...
+      end
 
-      ran = false
-      expect(env.with_text_domain { ran = true; :result }).to eq(:result)
-      expect(ran).to eq(true)
+      it "does not use the existing text domain even if it has been created" do
+        Puppet[:disable_i18n] = false
+        env.with_text_domain do; end
+
+        Puppet[:disable_i18n] = true
```
Comment (ciprianbadescu) [inline spec/unit/node/environment_spec.rb:574]: what is the purpose of these lines?

Comment (ciprianbadescu) [reply spec/unit/node/environment_spec.rb]: Is this a case that could happen for real? Having a config option (like `disable_i18n`) changed during the test (from `false` to `true`) make me believe we are testing something that will practically occur.

Comment (luchihoratiu) [reply spec/unit/node/environment_spec.rb]: After browsing the code a bit more carefully, I see what you mean about the likeability of this ever happening in a real situation and removed the test. Great catch! 🚀 Will try next time to question the use case a bit more before going for higher test coverage 😅 ..

**Review by ciprianbadescu — APPROVED** [review]

File: lib/puppet/configurer/plugin_handler.rb [hunk 2]
```diff
@@ -29,25 +29,27 @@ def download_plugins(environment)
...
-    if locales
-      locales_downloader = Puppet::Configurer::Downloader.new(
-        "locales",
-        Puppet[:localedest],
-        Puppet[:localesource],
-        Puppet[:pluginsignore] + " *.pot config.yaml",
-        environment
-      )
-      result += locales_downloader.evaluate
+    if Puppet[:disable_i18n] == false
```
Comment (joshcooper) [inline lib/puppet/configurer/plugin_handler.rb:32]: nit, comparing the boolean setting with true/false is redundant. Maybe just `unless Puppet[:disable_i18n]`?

File: lib/puppet/defaults.rb [hunk 3]
```diff
@@ -196,7 +196,7 @@ def self.initialize_default_settings!(settings)
       end
     },
     :disable_i18n => {
-      :default => false,
+      :default => true,
```
Comment (joshcooper) [inline lib/puppet/defaults.rb:199]: I generally try to keep commits into separate logic units, so one commit to make the locale pluginsync observe the setting and another to change the setting's default value.

**Review by GabrielNagy — APPROVED** [review]

**Comment by joshcooper** [comment]
jenkins please test this with servertests

**Review by joshcooper — APPROVED** [review]
LGTM, not sure if any `server` beaker tests need to change?

**Comment by Magisus** [comment]
Yes, if I'm understanding right, with this change, if someone _did_ re-enable i18n on an agent, locales would correctly be downloaded? If that is true, this looks fine to me.

**Comment by Magisus** [comment]
We don't have any i18n tests living in our repo, but I think there quite a few in this repo that will start failing with this, that either test it directly or were special-cased to handle Japanese. For example, https://github.com/puppetlabs/puppet/blob/fc4d42d974f55ee1bdf7d04aa974baa4ddd7353a/acceptance/tests/i18n/modules/puppet_describe.rb. We also might want to re-evaluate the utility of the [Japanese windows cell](https://github.com/puppetlabs/ci-job-configs/blob/master/jenkii/platform/projects/puppet-agent.yaml#L60) in the test matrix.

**Comment by joshcooper** [comment]
> I think there quite a few in this repo that will start failing

I think it didn't fail because the test is tagged as `medium`. Might be good to bump the risk level or at least manually run the test, and these too. Likely need to pass `puppet agent -t --no-disable_i18n`:

```
acceptance/lib/puppet/acceptance/i18ndemo_utils.rb:      language = enable_locale_language(master, language)
acceptance/tests/i18n/check_puppet_run_message.rb:      language = enable_locale_language(agent, language)
acceptance/tests/i18n/enable_option_disable_i18n.rb:        language = enable_locale_language(agent, language)
acceptance/tests/i18n/modules/puppet_agent.rb:    agent_language = enable_locale_language(agent, language)
acceptance/tests/i18n/modules/puppet_agent_cached_catalog.rb:    agent_language = enable_locale_language(agent, language)
acceptance/tests/i18n/modules/puppet_agent_with_multiple_environments.rb:    agent_language = enable_locale_language(agent, language)
acceptance/tests/i18n/modules/puppet_apply.rb:    agent_language = enable_locale_language(agent, language)
acceptance/tests/i18n/modules/puppet_apply_module_lang.rb:    agent_language = enable_locale_language(agent, language)
acceptance/tests/i18n/modules/puppet_describe.rb:    agent_language = enable_locale_language(agent, language)
acceptance/tests/i18n/modules/puppet_face.rb:    agent_language = enable_locale_language(agent, language)
acceptance/tests/i18n/modules/puppet_facts.rb:    agent_language = enable_locale_language(agent, language)
acceptance/tests/i18n/modules/puppet_resource.rb:    agent_language = enable_locale_language(agent, language)
acceptance/tests/utf8/utf8-recursive-copy.rb:  # for enable_locale_language
acceptance/tests/utf8/utf8-recursive-copy.rb:      if enable_locale_language(host, 'en_US').nil?
```

**Comment by luchihoratiu** [comment]
jenkins please test this on windows2019_ja-64a

**Comment by luchihoratiu** [comment]
jenkins please test this on windows2019_ja-64a

**Comment by luchihoratiu** [comment]
jenkins please test this on aix61-POWERfa,aix71-POWERfa,aix72-POWERfa,amazon6-64a,amazon7-64a,debian8-64a,debian8-32a,debian9-64a,debian9-32a,debian10-64a,fedora30-64a,fedora31-64a,fedora32-64a,osx1014-64a,osx1015-64a,redhat5-64a,centos6-64a,centos6-32a,centos8-64a,redhat7-64a,redhat7-POWERa,redhat7-AARCH64a,redhat8-64a,redhatfips7-64a,sles11-64a,sles11-32a,sles12-64a,sles12-POWERa,sles15-64a,solaris10-64a,solaris10-SPARCa,solaris11-64a,solaris11-SPARCa,solaris114-64a,ubuntu1404-64a,ubuntu1404-32a,ubuntu1604-64a,ubuntu1604-32a,ubuntu1604-POWERa,ubuntu1804-64a,ubuntu2004-64a,windows10ent-64a,windows10ent-32a,windows2008r2-64a,windows2012r2-64a,windows2019_ja-64a,windows2016-64a,windows2019-64a,windowsfips2012r2-64

**Comment by GabrielNagy** [comment]
jenkins please test this on aix61-POWERfa,aix71-POWERfa,aix72-POWERfa,amazon6-64a,amazon7-64a,debian9-64a,debian9-32a,debian10-64a,fedora30-64a,fedora31-64a,fedora32-64a,osx1014-64a,osx1015-64a,centos6-64a,centos6-32a,centos8-64a,redhat7-64a,redhat7-POWERa,redhat7-AARCH64a,redhat8-64a,redhatfips7-64a,sles12-64a,sles12-POWERa,sles15-64a,solaris11-64a,solaris11-SPARCa,solaris114-64a,ubuntu1604-64a,ubuntu1604-32a,ubuntu1604-POWERa,ubuntu1804-64a,ubuntu2004-64a,windows10ent-64a,windows10ent-32a,windows2012r2-64a,windows2019_ja-64a,windows2016-64a,windows2019-64a,windowsfips2012r2-64

**Comment by luchihoratiu** [comment]
jenkins please test this on aix61-POWERfa,aix71-POWERfa,aix72-POWERfa,amazon6-64a,amazon7-64a,debian9-64a,debian9-32a,debian10-64a,fedora30-64a,fedora31-64a,fedora32-64a,osx1014-64a,osx1015-64a,centos6-64a,centos6-32a,centos8-64a,redhat7-64a,redhat7-POWERa,redhat7-AARCH64a,redhat8-64a,redhatfips7-64a,sles12-64a,sles12-POWERa,sles15-64a,solaris11-64a,solaris11-SPARCa,solaris114-64a,ubuntu1604-64a,ubuntu1604-32a,ubuntu1604-POWERa,ubuntu1804-64a,ubuntu2004-64a,windows10ent-64a,windows10ent-32a,windows2012r2-64a,windows2019_ja-64a,windows2016-64a,windows2019-64a,windowsfips2012r2-64

**Review by joshcooper — APPROVED** [review]

**Pull request merged**